demisto_sdk/commands/doc_reviewer/tests/doc_reviewer_test.py

```diff
@@ -311,6 +320,166 @@ def test_failure_on_malformed_rns(self, pack):
+            A non-XSOAR-supported Pack with correct spelling.
+
+        When -
+            Running `doc-review` on a non-XSOAR-supported Pack with `--xsoar-only` flag set.
+
+        Then -
+            Ensure `doc-review` succeeds.
+        """
+
+        cmd_args: List[str] = [
+            "--input",
+            non_supported_pack.path,
+        ]
+
+        result = self.run_doc_review_cmd(cmd_args)
+
+        assert result.exit_code == self.CommandResultCode.SUCCESS.value
+
+    def test_valid_multiple_supported_packs(self, supported_packs: List[Pack]):
+
+        """
+        Given -
+            2 XSOAR-supported Packs with correct spelling.
+
+        When -
+            Running `doc-review` on XSOAR-supported Pack with `--xsoar-only` flag set.
+
+        Then -
+            Ensure `doc-review` succeeds.
+        """
+
+        cmd_args: List[str] = ["--xsoar-only"]
+        for pack in supported_packs:
+            cmd_args.append("--input")
+            cmd_args.append(pack.path)
+
+        result = self.run_doc_review_cmd(cmd_args)
+
+        assert result.exit_code == self.CommandResultCode.SUCCESS.value
+
+    def test_invalid_non_supported_pack(self, non_supported_pack_mispelled: Pack):
+
+        """
+        Given -
+            A non-XSOAR-supported Pack with incorrect spelling.
+
+        When -
+            Running `doc-review` on a non-XSOAR-supported Pack with `--xsoar-only` flag set.
+
+        Then -
+            Ensure `doc-review` succeeds.
+        """
+
+        cmd_args: List[str] = [
+            "--input",
+            non_supported_pack_mispelled.path,
+        ]
+
+        result = self.run_doc_review_cmd(cmd_args)
+
+        assert result.exit_code == self.CommandResultCode.SUCCESS.value
+
+    def test_invalid_supported_pack(self, supported_pack_mispelled: Pack):
+
+        """
+        Given -
+            A XSOAR-supported Pack with incorrect spelling.
+
+        When -
+            Running `doc-review` on a non-XSOAR-supported Pack with `--xsoar-only` flag set.
+
+        Then -
+            Ensure `doc-review` succeeds.
+        """
+
+        cmd_args: List[str] = [
+            "--input",
+            supported_pack_mispelled.path,
+        ]
+
+        result = self.run_doc_review_cmd(cmd_args)
+
+        assert result.exit_code == self.CommandResultCode.FAIL.value
+
+    def test_invalid_mix_packs(self, mix_invalid_packs: List[Pack]):
+
+        """
+        Given -
+            2 Packs, one community, one XSOAR-supported with incorrect spelling.
+
+        When -
+            Running `doc-review` on both Packs with `--xsoar-only` flag set.
+
+        Then -
+            Ensure `doc-review` fails.
+        """
+
+        cmd_args: List[str] = ["--xsoar-only"]
+        for pack in mix_invalid_packs:
+            cmd_args.append("--input")
+            cmd_args.append(pack.path)
+
+        result = self.run_doc_review_cmd(cmd_args)
+
+        assert result.exit_code == self.CommandResultCode.FAIL.value
```

Comment:
can we also assert that the only one checked is the XSOAR supported one?

Reply:
@DinaMeylakh - This test case's intention is to ensure that the exit code in this test is non-zero.

I wrote `TestDocReviewPrinting::test_printing_skip_non_xsoar_supported_file` to ensure that we're checking only the supported file.

Reply:
@DinaMeylakh - This test case's intention is to ensure that the exit code in this test is non-zero. 

In [`7c2c684` (#2660)](https://github.com/demisto/demisto-sdk/pull/2660/commits/7c2c6845dc075daef93be4e0514e29648c02bc8f), I added `TestDocReviewPrinting::test_printing_skip_non_xsoar_supported_file` to ensure that we're checking only the supported file.